cockatrice/src/userinfobox.cpp

```diff
@@ -140,3 +162,126 @@ void UserInfoBox::processResponse(const Response &r)
+    
+    PendingCommand *pend = client->prepareSessionCommand(cmd);
+    connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(processEditResponse(const Response &)));
+    
+    client->sendCommand(pend);
+}
+
+void UserInfoBox::actPassword()
+{
+    DlgEditPassword dlg(this);
+    if(!dlg.exec())
+        return;
+
+    Command_AccountPassword cmd;
+    cmd.set_old_password(dlg.getOldPassword().toStdString());
+    cmd.set_new_password(dlg.getNewPassword().toStdString());
+    
+    PendingCommand *pend = client->prepareSessionCommand(cmd);
+    connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(processPasswordResponse(const Response &)));
+    
+    client->sendCommand(pend);
+}
+
+void UserInfoBox::actAvatar()
+{
+    DlgEditAvatar dlg(this);
+    if(!dlg.exec())
+        return;
+
+    Command_AccountImage cmd;
+    cmd.set_image(dlg.getImage().data(), dlg.getImage().size());
+    
+    PendingCommand *pend = client->prepareSessionCommand(cmd);
+    connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(processAvatarResponse(const Response &)));
+    
+    client->sendCommand(pend);
+}
+
+void UserInfoBox::processEditResponse(const Response &r)
+{
+    switch (r.response_code()) {
+        case Response::RespOk:
+            updateInfo(nameLabel.text());
+            QMessageBox::information(this, tr("Information"), tr("User information updated."));
+            break;
+        case Response::RespFunctionNotAllowed:
+            QMessageBox::critical(this, tr("Error"), tr("This server does not permit you to update your user informations."));
+            break;
+        case Response::RespInternalError:
+        default:
+            QMessageBox::critical(this, tr("Error"), tr("An error occured while trying to update your user informations."));
+            break;
+    }
+}
+
+void UserInfoBox::processPasswordResponse(const Response &r)
+{
+    switch (r.response_code()) {
+        case Response::RespOk:
+            QMessageBox::information(this, tr("Information"), tr("Password changed."));
+            break;
+        case Response::RespFunctionNotAllowed:
+            QMessageBox::critical(this, tr("Error"), tr("This server does not permit you to change your password."));
+            break;
+        case Response::RespPasswordTooShort:
+            QMessageBox::critical(this, tr("Error"), tr("The new password is too short."));
+            break;
+        case Response::RespWrongPassword:
+            QMessageBox::critical(this, tr("Error"), tr("The old password is incorrect."));
+            break;
+        case Response::RespInternalError:
+        default:
+            QMessageBox::critical(this, tr("Error"), tr("An error occured while trying to update your user informations."));
+            break;
+    }
+}
+
+void UserInfoBox::processAvatarResponse(const Response &r)
+{
+    switch (r.response_code()) {
```

Comment:
The database is using a BLOB to store the image, than in mysql means 64kbs.. probably not enough for a big image. That's why currently the user-chosen image is resized down to 400x200 and sent in jpg format.
I meant to address this making mysql use a MEDIUMBLOB (16mbs) and adding the relevant server-side checks and settings.
This would touch some more files and add more code, so i was planning on fixing it on a following PR.